app/src/main/java/com/nutomic/syncthingandroid/model/Completion.java

```diff
@@ -0,0 +1,87 @@
+package com.nutomic.syncthingandroid.model;
+
+import java.util.HashMap;
+
+/**
+ * This class caches remote folder and device synchronization
+ * completion indicators defined in {@link CompletionInfo#CompletionInfo}
+ * according to syncthing's REST "/completion" JSON result schema.
+ * Completion model of syncthing's web UI is completion[deviceId][folderId]
+ */
+public class Completion {
+
+    HashMap<String, HashMap<String, CompletionInfo>> deviceFolderMap =
+        new HashMap<String, HashMap<String, CompletionInfo>>();
+
+    /**
+     * Adds a device to the cache model if it does not exist.
+     */
+    private void addDevice(String deviceId) {
+        if (!deviceFolderMap.containsKey(deviceId))
+            deviceFolderMap.put(deviceId, new HashMap<String, CompletionInfo>());

```

Comment:
You should be consistent about braces, like `if (...) {}`. I'd say it's better to always have braces, as that prevents errors and makes clear what's happening.

Reply:
Ok, good idea.